### DISCUSSION TITLE
How to initialize reservoir pressure?

### DISCUSSION THREAD

**Discussion by wdmhouston** [GEOS-DEV/GEOS#1436] [Q&A] [answered]
Hi,

    I find the following example code in CO2 injection example,

    <FieldSpecification
      name="initialPressure"
      initialCondition="1"
      setNames="{ all }"
      objectPath="ElementRegions/Reservoir"
      fieldName="pressure"
      scale="1.25e7"/>

     Does this mean the initial pressure of each cell is 1x1.25e7 bar?

     Does GEOSX in current release support pressure from a GRDECL file (or other format), or specify an initial pressure at certain depth(automatically equilibrate for the whole model)?

Thanks and Regards,

wdm

**Comment by francoishamon** [comment, marked as answer]
Hello, 
 This means that the initial pressure is uniform, with a value in each cell equal to 1.25 x 10^7 ***Pa***. All the pressure units that are in GEOSX (reservoir, wells) are in Pa.

Currently we do not support the `PRESSURE` input from the GRDECL file. Here is what you have to do to initialize the reservoir for pressure (and possibly for the other variables as well if you have, say, a gas-oil contact or something like that):
```
<FieldSpecification
      name="initialPressure"
      initialCondition="1"
      setNames="{ all }"
      objectPath="ElementRegions/Reservoir"
      fieldName="pressure"
      scale="1"
      functionName="initialPressureFunction"/>

    <FieldSpecification
      name="initialComposition_oil"
      initialCondition="1"
      setNames="{ all }"
      objectPath="ElementRegions/Reservoir"
      fieldName="globalCompFraction"
      component="0"
      scale="1.0"
      functionName="initialOilCompFunction"/>

    <FieldSpecification
      name="initialComposition_water"
      initialCondition="1"
      setNames="{ all }"
      objectPath="ElementRegions/Reservoir"
      fieldName="globalCompFraction"
      component="1"
      scale="1.0"
      functionName="initialWaterCompFunction"/>

```
and then in the `Functions` block of the xml file, you can do the following to initialize the pressure and the compositions, as a function of the position of the center of each cell:
```
<Functions>

    <TableFunction
      name="initialPressureFunction"
      inputVarNames="{elementCenter}"
      coordinateFiles="{xLin.geos,yLin.geos,zLin.geos}"
      voxelFile="pres.geos"
      interpolation="linear" />

    <TableFunction
      name="initialOilCompFunction"
      inputVarNames="{elementCenter}"
      coordinateFiles="{xLin.geos,yLin.geos,zLin.geos}"
      voxelFile="comp_o.geos"
      interpolation="linear" />

    <TableFunction
      name="initialWaterCompFunction"
      inputVarNames="{elementCenter}"
      coordinateFiles="{xLin.geos,yLin.geos,zLin.geos}"
      voxelFile="comp_w.geos"
      interpolation="linear" />
    
  </Functions>
```
If we take the example of the pressure above, you can use the `pres.geos` file to define the pressure at the locations given by the files `xLin.geos`, `yLin.geos`, and `zLin.geos`. Then, internally, GEOSX uses the center of each cell to interpolate in the table and assign a pressure value. 

What you have to do is to compute externally the hydrostatic pressure, fill the `pres.geos` file with the hydrostatic pressure values, and fill the `xLin.geos`, `yLin.geos`, and `zLin.geos` with the coordinates of your reservoir. I attached example files (I had to remove the .geos extension to upload, you will have to put them back)
[zLin.txt](https://github.com/GEOSX/GEOSX/files/6549492/zLin.txt)
[yLin.txt](https://github.com/GEOSX/GEOSX/files/6549493/yLin.txt)
[xLin.txt](https://github.com/GEOSX/GEOSX/files/6549494/xLin.txt)
[comp_w.txt](https://github.com/GEOSX/GEOSX/files/6549495/comp_w.txt)
[comp_o.txt](https://github.com/GEOSX/GEOSX/files/6549496/comp_o.txt)
[pres.txt](https://github.com/GEOSX/GEOSX/files/6549497/pres.txt)

We will start working on an internal initialization procedure, but it is not ready yet.
